### PR TITLE
When there are too many hits, provide a link to Focus Mode

### PR DESCRIPTION
**Pull request by jonbell-lot23**
**Old:**
image

**New:**
image

--

Additional notes about adding conditionals:

Loom:
https://www.loom.com/share/c46da8f282e644aa884ab0c08220a251

Following up on [DES-246](https://linear.app/replay/issue/DES-246/consider-making-print-statements-with-200-hits-interactable), we want to provide people with a way to address things when there are too many hits. The best UX is to provide links to both Focus Mode and conditionals, but the conditional work may not be feasible yet. To discuss as a team, but in the meantime this PR can go live.

### CODE REVIEW

**Comment by jonbell-lot23** [comment]
Hello @bvaughn ! I wrote this code in order to add Focus Mode and conditionals as a proof of concept Loom video.

Then I removed the conditionals link and was going to commit the Focus Mode link but there's an error in the code. Feel free to fork off this or just rewrite it yourself. Keeping it around in case it's useful.

**Comment by jonbell-lot23** [comment]
This is ready for review!

* Added a Focus Mode link to the error state
* Changed the default "too many hits" from "-" to "10k+"

**Review by jasonLaster — COMMENTED** [review]
Looks good, but will let @bvaughn do a proper review. The main thing for me would be if `enterFocusMode` should be in context, dispatch, or something else...

File: packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx [hunk 1]
```diff
@@ -87,7 +89,12 @@ function PointPanel(props: ExternalProps) {
   );
 
   return (
-    <PointPanelWithHitPoints {...props} hitPoints={hitPoints} hitPointStatus={hitPointStatus} />
+    <PointPanelWithHitPoints
+      {...props}
+      hitPoints={hitPoints}
+      hitPointStatus={hitPointStatus}
+      enterFocusMode={enterFocusMode}
```
Comment (bvaughn) [inline packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx:96]: nit: Alpha-sort props

File: packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx [hunk 2]
```diff
@@ -97,6 +104,7 @@ function PointPanelWithHitPoints({
   hitPointStatus,
   point,
   setLinePointState,
+  enterFocusMode,
```
Comment (bvaughn) [inline packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx:107]: nit: Alpha-sort params

File: packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx [hunk 3]
```diff
@@ -340,7 +348,11 @@ function PointPanelWithHitPoints({
       <div className={styles.EditableContentWrapperRow}>
         {showTooManyPointsMessage ? (
           <div className={styles.ContentWrapperTooManyPoints}>
-            Use Focus Mode to reduce the number of hits.
+            Use{""}
+            <span style={{ textDecoration: "underline" }} onClick={enterFocusMode}>
```
Comment (bvaughn) [inline packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx:352]: For consistency with the rest of the app, styles like this should be in the CSS module, not inline. (We only use inline styles for _dynamic_ things.)

File: packages/replay-next/src/contexts/FocusContext.tsx [hunk 4]
```diff
@@ -30,6 +30,7 @@ export type FocusContextType = {
   range: TimeStampedPointRange | null;
   rangeForDisplay: TimeStampedPointRange | null;
   update: (value: Range | null, debounce: boolean) => void;
+  enterFocusMode: () => void;
```
Comment (bvaughn) [inline packages/replay-next/src/contexts/FocusContext.tsx:33]: Alpha sort.

File: packages/replay-next/src/contexts/FocusContext.tsx [hunk 5]
```diff
@@ -122,6 +123,7 @@ export function FocusContextRoot({ children }: PropsWithChildren<{}>) {
       rangeForDisplay: range,
       range: deferredRange,
       update: updateFocusRange,
+      enterFocusMode: () => {},
```
Comment (bvaughn) [inline packages/replay-next/src/contexts/FocusContext.tsx:126]: Alpha sort.

File: packages/replay-next/src/utils/testing.tsx [hunk 6]
```diff
@@ -110,6 +110,7 @@ export async function renderFocused(
     range: null,
     rangeForDisplay: null,
     update: jest.fn(),
+    enterFocusMode: () => {},
```
Comment (bvaughn) [inline packages/replay-next/src/utils/testing.tsx:113]: Alpha sort.

File: src/ui/components/FocusContextReduxAdapter.tsx [hunk 7]
```diff
@@ -45,9 +45,12 @@ export default function FocusContextReduxAdapter({ children }: PropsWithChildren
       isTransitionPending: isPending,
       range: deferredFocusRegion ? rangeForFocusRegion(deferredFocusRegion) : null,
       rangeForDisplay: focusRegion ? rangeForFocusRegion(focusRegion) : null,
+      enterFocusMode: () => {
+        dispatch(enterFocusMode());
+      },
       update,
     };
-  }, [deferredFocusRegion, isPending, focusRegion, update]);
+  }, [deferredFocusRegion, isPending, focusRegion, update, dispatch]);
```
Comment (bvaughn) [inline src/ui/components/FocusContextReduxAdapter.tsx:53]: Alpha sort.

File: packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx (hunk 3 above)
Comment (jonbell-lot23) [reply packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx:352]: Ah ha, that makes sense. Thank you, will address!

Comment (markerikson) [reply packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx:352]: That would be my fault! :)

**Comment by jonbell-lot23** [comment]
Addressed alphabetising and inline css

File: packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx (hunk 2 above)
Comment (bvaughn) [inline packages/replay-next/components/sources/log-point-panel/LogPointPanel.tsx:107]: Alpha sort 😉

File: src/ui/components/FocusContextReduxAdapter.tsx [hunk 8]
```diff
@@ -45,9 +45,12 @@ export default function FocusContextReduxAdapter({ children }: PropsWithChildren
       isTransitionPending: isPending,
       range: deferredFocusRegion ? rangeForFocusRegion(deferredFocusRegion) : null,
       rangeForDisplay: focusRegion ? rangeForFocusRegion(focusRegion) : null,
+      enterFocusMode: () => {
+        dispatch(enterFocusMode());
+      },
```
Comment (bvaughn) [inline src/ui/components/FocusContextReduxAdapter.tsx:50]: Alpha sort 😄

**Review by bvaughn — COMMENTED** [review]
Alpha-sorting suggestions are kind of nits, but I stick with them because:
1. They reduce the likelihood of merge conflicts
2. They make code easier to read/scan at a glance

**Review by bvaughn — APPROVED** [review]

**Comment by jonbell-lot23** [comment]
> Alpha-sorting suggestions are kind of nits, but I stick with them because:
> 
> 1. They reduce the likelihood of merge conflicts
> 2. They make code easier to read/scan at a glance

Totally get it, and I agree. Thank you for helping me improve!

I addressed the two most recent issues.

**Pull request merged**